sklearn/tests/test_kernel_approximation.py

```diff
@@ -20,6 +21,33 @@
 Y /= Y.sum(axis=1)[:, np.newaxis]
 
 
+@pytest.mark.parametrize('X', [X, csr_matrix(X)])
+@pytest.mark.parametrize('Y', [Y, csr_matrix(Y)])
+@pytest.mark.parametrize('gamma', [0.1, 1, 2.5])
+@pytest.mark.parametrize('degree', [1, 2, 4])
+@pytest.mark.parametrize('coef0', [0, 1, 4])
+def test_polynomial_sampler(X, Y, gamma, degree, coef0):
+    # test that PolynomialSampler approximates polynomial
+    # kernel on random data
+
+    # compute exact kernel
+    kernel = (gamma * np.dot(X, Y.T) + coef0)**degree
```

Comment:
Better use `metrics.pairwise.polynomial_kernel` as `np.dot` will fail for sparse matrices. And `polynomial_kernel` internally uses ` sklearn.utils.extmath.safe_sparse_dot` which takes care of the different cases.

Reply:
Done, thank you very much for the tips!